lib/resty/etcd/v3.lua

```diff
@@ -249,8 +265,9 @@ function refresh_jwt_token(self, timeout)
             password     = self.password,
         }
     }
-    local res, err = _request_uri(self, 'POST',
-                                  choose_endpoint(self).full_prefix .. "/auth/authenticate",
+    local endpoint = choose_endpoint(self)
```

Comment:
could we put this into function `_request_uri` so that we don't need to invoke everywhere ?

Reply:
let me try

Reply:
not easy, because the third parameter `uri` of the function `_request_uri`  needs to be passed in by the invoke function, such as
https://github.com/api7/lua-resty-etcd/blob/f2abd8b99629c1c8b133aa25e6c7c20a255d0d4a/lib/resty/etcd/v3.lua#L996-L997
https://github.com/api7/lua-resty-etcd/blob/f2abd8b99629c1c8b133aa25e6c7c20a255d0d4a/lib/resty/etcd/v3.lua#L1003-L1004
so the invoke function need to `choose endpoint` and choose to use `endpoint.full_prefix` or `endpoint.http_host`.

Reply:
OK